content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,412 @@ preview: true
+
+```prisma file=schema.prisma highlight=4,5;add
+model Post {
+  id          Int     @id @default(autoincrement())
+  title       String
+  authorName  String  @default("anonymous")
+  author      User?   @relation(fields: [authorId], references: [id], onDelete: SetDefault, onUpdate: SetDefault)
+  authorId    Int?
+}
+
+model User {
+  username  String  @id
+  posts     Post[]
+}
+```
+
+##### Result of using `SetDefault`
+
+When deleting a `User`, its existing authored posts will now be authored by a `User` with `id` "anonymous". 
+
+When the `username` of a `User` changes, its existing authored posts will now be authored by `User` with `username` "anonymous".
+
+## Upgrade paths
+
+There are a couple of paths you can take when upgrading which will give different results depending on the desired outcome.
+
+If you currently use the migration workflow, you can run an introspection to check how the defaults are reflected in your schema. You can then manually update your database if you need to. 
+
+You can also decide to skip checking the the defaults and run a migration to update your database with the [new default values](#referential-action-defaults).
+
+The following assumes you have upgraded to 2.26.0 and [enabled the preview feature flag](#enable-the-referentialactions-preview-feature).
+
+### Using Introspection
+
+If you [Introspect](../introspection) your database, the referential actions configured at the database level will be reflected in your Prisma Schema. If you have been using Prisma Migrate or `prisma db push` to manage the database schema, these are likely to be the [<=2.25.0 default values](#the-2250-default-referential-actions). 
+
+When you run an Introspection, Prisma compares all the foreign keys in the database with the schema, if the SQL statements `ON DELETE` and `ON UPDATE` do **not** match the default values, they will be explicitly set in the schema file.
+
+After introspecting, you can review the non-default clauses in your schema. The most important clause to review is `onDelete`, which defaults to `Cascade` in 2.25.0 and earlier.
+
+<Admonition type="alert">
+
+If you are using either the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods, **[cascading deletes](#how-to-use-cascading-deletes) will now be performed** as the `referentialAction` preview feature **removed the safety net in Prisma Client that previously prevented cascading deletes at runtime**. Be sure to check your code and make any adjustments accordingly.
+
+</Admonition>
+
+Make sure you are happy with every case of `onDelete: Cascade` in your schema - if not, modify your schema and push/migrate, or modify the underlying database if you are using an introspection-only workflow.
```

Comment:
```suggestion
Make sure you are happy with every case of `onDelete: Cascade` in your schema. If not, either:

* Modify your Prisma schema and `db push` or `dev migrate` to change the database_or_
* Manually update the underlying database if you use an introspection-only workflow
```

Reply:
Split this out into two points so that you can skim and see the scenario that fits you

Reply:
Yah, just copied your copy, like the pro I am. 🤦‍♂️